dagger-functions/src/test/java/io/odpf/dagger/functions/udfs/python/PythonUdfManagerTest.java

```diff
@@ -0,0 +1,162 @@
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+        when(pythonUdfConfig.getPythonArchives()).thenReturn("/path/to/file.txt");
+        when(pythonUdfConfig.getPythonRequirements()).thenReturn("requirements.txt");
+        when(pythonUdfConfig.getPythonArrowBatchSize()).thenReturn(10000);
+        when(pythonUdfConfig.getPythonBundleSize()).thenReturn(100000);
+        when(pythonUdfConfig.getPythonBundleTime()).thenReturn(1000L);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(configuration, times(1)).setString("python.archives", "/path/to/file.txt");
+        verify(configuration, times(1)).setString("python.requirements", "requirements.txt");
+        verify(configuration, times(1)).setInteger("python.fn-execution.arrow.batch.size", 10000);
+        verify(configuration, times(1)).setInteger("python.fn-execution.bundle.size", 100000);
+        verify(configuration, times(1)).setLong("python.fn-execution.bundle.time", 1000);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterFirstUdf);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterSecondUdf);
+    }
+
+    @Test
+    public void shouldNotRegisterConfigIfNotSet() throws IOException {
+        String pathFile = getPath("python_udf.zip");
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(configuration, times(0)).setString("python.archives", "/path/to/file.txt");
+        verify(configuration, times(0)).setString("python.requirements", "requirements.txt");
+    }
+
+    @Test
+    public void shouldRegisterPythonUdfFromPyFile() throws IOException {
+        String pathFile = getPath("test_udf.py");
+        String sqlRegisterUdf = "CREATE TEMPORARY FUNCTION TEST_UDF AS 'test_udf.test_udf' LANGUAGE PYTHON";
+
+        when(tableEnvironment.getConfig()).thenReturn(tableConfig);
+        when(tableConfig.getConfiguration()).thenReturn(configuration);
+        when(pythonUdfConfig.getPythonFiles()).thenReturn(pathFile);
+
+        PythonUdfManager pythonUdfManager = new PythonUdfManager(tableEnvironment, pythonUdfConfig);
+        pythonUdfManager.registerPythonFunctions();
+
+        verify(configuration, times(1)).setString("python.files", pathFile);
+        verify(tableEnvironment, times(1)).executeSql(sqlRegisterUdf);
+    }
+
+    @Test
+    public void shouldRegisterPythonUdfFromPyAndZipFile() throws IOException {
```

Comment:
can we add more tests around, different types of filename and extensions? what happens with extra spaces in python.files config?

Reply:
there's already test using .txt extension. should we add test too for extra spaces on given config? didn't see this on other test (StreamConfigTest)

Reply:
what happens if there are other file formats inside zip

Reply:
inside zip it will only get .py format. else will be skipped

Reply:
yes,  but can we add test for that?

Reply:
ok, will add new file inside zip. so inside zip files will be 2 py files and 1 txt file. the test will be verify that only execute 2 py files.